### PR TITLE
Circe improvements

### PR DESCRIPTION
**Pull request by shaun-whitely**
- Remove `personToJson` and `jsonToPerson` exercises and just go straight to building encoders and decoders
- Remove `forProductN` exercises in the interest of time as these aren't essential to understanding encoders and decoders
- Talk about using the companion object to hold implicits
- Touch on type classes briefly

### CODE REVIEW

File: src/main/scala/com/reagroup/exercises/circe/CirceExercises.scala [hunk 1]
```diff
@@ -64,30 +47,69 @@ object CirceExercises {
     """
   }
 
+  case class Person(name: String, age: Int)
+
   /**
-    * Create an `Encoder` instance for `Person` by implementing the `apply` method below.
+    * Defining encoders and decoders in the companion object means that Scala will always be able to find them.
```
Comment (bhargysh) [inline src/main/scala/com/reagroup/exercises/circe/CirceExercises.scala:53]: Nice I like this 👏

File: src/main/scala/com/reagroup/exercises/circe/CirceExercises.scala [hunk 2]
```diff
@@ -64,30 +47,69 @@ object CirceExercises {
...
   /**
-    * There is an alternate way to construct an `Encoder` instance,
-    * by using `Encoder.forProduct2`.
+    * Scala will look for an implicit `Encoder[Person]` in the following places:
     *
-    * This may sometimes be simpler than using `Json.obj`.
+    * - The current scope (current method, class, file)
+    * - Imports
+    * - The companion object of `Encoder`
+    * - The companion object of `Person` (bingo!)
```
Comment (bhargysh) [inline src/main/scala/com/reagroup/exercises/circe/CirceExercises.scala:100]: 💪

File: src/main/scala/com/reagroup/exercises/circe/CirceExercises.scala [hunk 3]
```diff
@@ -146,46 +162,43 @@ object CirceExercises {
...
     implicit val personDecoder: Decoder[Person] = new Decoder[Person] {
-      override def apply(cursor: HCursor): Result[Person] = ???
+      override def apply(cursor: HCursor): Result[Person] = for {
+        name <- cursor.get[String]("name")
+        age <- cursor.get[Int]("age")
+      } yield Person(name, age)
     }
+    // note: a lot of boilerplate can be removed. Try pressing alt-enter with your
+    // cursor over "new Decoder[Person]" above. This works because Decoder is a trait with
+    // a single abstract method.
```
Comment (bhargysh) [inline src/main/scala/com/reagroup/exercises/circe/CirceExercises.scala:172]: Super helpful comments 💯

**Comment by shaun-whitely** [comment]
🤦  I committed the answers. Fixed now.

**Pull request merged**